### PR TITLE
Fixing sentence-transformers CACHE value for docker + functools (docker needs Py3.8)

### PR DESCRIPTION
**Pull request by Narsil**

### CODE REVIEW

File: api-inference-community/docker_images/common/app/main.py [hunk 1]
```diff
@@ -76,6 +76,11 @@ async def startup_event():
 
     # Link between `api-inference-community` and framework code.
     app.get_pipeline = get_pipeline
+    try:
+        get_pipeline()
+    except Exception:
+        # We can fail so we can show exception later.
+        pass
```
Comment (osanseviero) [inline api-inference-community/docker_images/common/app/main.py:83]: Could you please explain why this is needed here and also below?

File: api-inference-community/docker_images/sentence_transformers/Dockerfile [hunk 2]
```diff
@@ -5,6 +5,7 @@ LABEL maintainer="Omar <omar@huggingface.co>"
 # RUN apt-get update -y && apt-get install libXXX -y
 
 COPY ./requirements.txt /app
+RUN pip install --no-cache-dir torch==1.9
```
Comment (osanseviero) [inline api-inference-community/docker_images/sentence_transformers/Dockerfile:8]: Why not have this in requirements?

**Review by osanseviero — APPROVED** [review]
Looks good, thanks!

File: api-inference-community/docker_images/sentence_transformers/Dockerfile (hunk 2 above)
Comment (Narsil) [reply api-inference-community/docker_images/sentence_transformers/Dockerfile:8]: It's better for cache.

Having it before `requirements.txt` means as long torch version is not required to be updated in `requirements.txt` then you will hit cache and you won't have to redownload torch all the time.

Comment (osanseviero) [reply api-inference-community/docker_images/sentence_transformers/Dockerfile:8]: Sounds good, thanks!

**Pull request merged**